var/spack/repos/builtin/packages/legion/package.py

```diff
@@ -350,11 +353,10 @@ def cmake_args(self):
             options.append(from_variant("Legion_HIP_TARGET", "hip_target"))
             options.append(from_variant("Legion_HIP_ARCH", "amdgpu_target"))
             options.append(from_variant("Legion_HIJACK_HIP", "hip_hijack"))
-            options.append(self.define("HIP_PATH", "{0}/hip".format(spec["hip"].prefix)))
-            if "^hip@:5.7" in spec:
+            if "@23.03.0:23.12.0" in spec:
                 options.append(self.define("HIP_PATH", "{0}/hip".format(spec["hip"].prefix)))
-            elif "^hip@6.0:" in spec:
-                options.append(self.define("HIP_PATH", "{0}".format(spec["hip"].prefix)))
+            else:
+                options.append(self.define("ROCM_PATH", "{0}".format(spec["hip"].prefix)))
```

Comment:
FYI. As we update the core, we are also encouraging use of Python f-strings in packages (see https://docs.python.org/3/tutorial/inputoutput.html).

Reply:
In this case, you can even avoid the format altogether
```suggestion
                options.append(self.define("ROCM_PATH",  spec["hip"].prefix))
```